### PR TITLE
Checkrelease snippet for Vagrantfile.  This prevents folks from deplo…

### LINKED ISSUE #18079: make salt blow up if release binaries are out of sync
Salt provisioners need to be in sync w/ binaries, otherwise `vagrant up` can fail .

I'm thinking we should be able to have salt confirm that the files in the binary are in sync with whats in the salt provisioner sources... The error message coule be fast and obvious enough that people could disable it if they wanted... for the typical use case, where you pull source down

otherwise you get something like this: 

```
[root@kubernetes-minion-1 vagrant]# diff /vagrant/cluster/saltbase/salt/kube-node-unpacker/init.sls /srv/salt/kube-node-unpacker/init.sls
4d3
<     - makedirs: True
```

### PR DESCRIPTION
**Pull request by jayunit100**
@mikedanese @derekwaynecarr  ... here's a quick way to do it.  Let me know if you have  a better way.
fixes #18079

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -7,16 +7,52 @@ VAGRANTFILE_API_VERSION = "2"
...
+    ./cluster/kube-up.sh
+    }
+    if ARGV.first == "up" && ENV['USING_KUBE_SCRIPTS'] != 'true'
+        return false, err
+    end
+    return true, nil
+end
+
+def binarycheck
+  if ARGV.first == "up"
```
Comment (mikedanese) [inline Vagrantfile]: if is unnecessary given previous check.

**Comment by jayunit100** [comment]
Ok @mikedanese 
- Updated PR according to conversation in #18079 (env var + warning).
- cleaned error checking behaviour so its all done w/ functions.

File: Vagrantfile [hunk 2]
```diff
@@ -7,16 +7,52 @@ VAGRANTFILE_API_VERSION = "2"
...
+    version="_output/release-stage/full/kubernetes/version"
+    
+    # make sure there is a release...
+    err="No kube release found ! If running from a clone of the git repo, please run 'make quick-release'."
+    return false, err if ! File.exist?(version)
```
Comment (mikedanese) [inline Vagrantfile]: This will always fail if I download a release tar using https://get.k8s.io/ or the github releases tab.

File: Vagrantfile [hunk 3]
```diff
@@ -7,16 +7,52 @@ VAGRANTFILE_API_VERSION = "2"
...
+
+    # make sure release matches source...
+    gitcommit = `git log -1 --abbrev-commit --format="%h"`.gsub("\n","")
+    releasecommit = `cat #{version}`
+    checkbinary = (ENV['IGNORE_BINARY_CHECK']  == nil )
```
Comment (mikedanese) [inline Vagrantfile]: non-descriptive env var name.

**Comment by mikedanese** [comment]
Why is this vagrant specific? Shouldn't this be factored to work with all kube-ups?

File: Vagrantfile (hunk 2 above)
Comment (jayunit100) [inline Vagrantfile]: Ohhhh I forgot about that use case.  Any clever generalized ideas on what to look for?

Comment (jayunit100) [inline Vagrantfile]: ah i see, there is a good way to check  in provision-master.sh

Comment (mikedanese) [inline Vagrantfile]: Please don't push this into provisioning code. Can you factor this to not be kube-up generic?

**Comment by jayunit100** [comment]
All kube-ups ? I kinda thought  so also,   but.. what about gke/juju/ ?

**Comment by mikedanese** [comment]
if the enter through ./cluster/kube-up there shouldn't be a problem, right?

**Comment by jayunit100** [comment]
well, do they even need a release at all ?
i guess even if they dont, they can override the check .
the majority of kube-ups do need a release i assume

**Comment by jayunit100** [comment]
seeing as how its going to be more complicated for other tools (i.e. gce, non dev releases), and that this is a minor improvement, im  closing for now...  

I think in the future maybe there will be a salt/ansible/go solution to this (i.e. when server spins up, send it a version arg or something).

**Pull request closed without merging**